syntax = "proto3";

package books;

service BookService {
  rpc GetAll (GetAllRequest) returns (GetAllResponse);
  rpc GetOne (GetOneRequest) returns (GetOneResponse);
  rpc PostOne (PostOneRequest) returns (PostOneResponse);
  rpc PostList (PostListRequest) returns (PostListResponse);
}

message Book {
  string id = 1;
  string title = 2;
  string author = 3;
  string isbn = 4;
  string description = 5;
  string price = 6;
}

message GetAllRequest {}

message GetAllResponse {
  repeated Book books = 1;
}

message GetOneRequest {
  string id = 1;
}
message GetOneResponse {
  Book book = 1;
}

message PostOneRequest {
  Book book = 1;
}
message PostOneResponse {
  Book book = 1;
}

message PostListRequest {
  repeated Book books = 1;
}
message PostListResponse {
  repeated Book books = 1;
}
